File: BACKEND/src/main/java/com/ROOMIFY/Roomify/model/Shop.java
package com.ROOMIFY.Roomify.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "shops")
@Getter
@Setter
public class Shop {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    
    @Column(columnDefinition = "TEXT")
    private String description;
    
    private String address;
    private double latitude;
    private double longitude;
    
    private String contactPhone;
    private String contactEmail;
    private String websiteUrl;
    
    @Column(name = "owner_id")
    private Long ownerId;
    
    @ElementCollection(fetch = FetchType.EAGER)
    @CollectionTable(name = "shop_images", joinColumns = @JoinColumn(name = "shop_id"))
    @Column(name = "image")
    private List<String> images = new ArrayList<>();
    
    private String logoUrl;
    private String videoUrl;
    private boolean hasVideo;
    
    private LocalDateTime createdAt = LocalDateTime.now();
    private LocalDateTime updatedAt = LocalDateTime.now();

    public Shop() {}
}

File: FRONTEND/shared/src/commonMain/kotlin/org/com/model/Shop.kt
package org.com.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import org.com.network.ApiClient

@Serializable
data class Shop(
    @SerialName("id")
    val id: Long? = null,
    @SerialName("name")
    val name: String = "",
    @SerialName("description")
    val description: String = "",
    @SerialName("address")
    val address: String? = null,
    @SerialName("latitude")
    val latitude: Double = 0.0,
    @SerialName("longitude")
    val longitude: Double = 0.0,
    @SerialName("contactPhone")
    val contactPhone: String? = null,
    @SerialName("contactEmail")
    val contactEmail: String? = null,
    @SerialName("websiteUrl")
    val websiteUrl: String? = null,
    @SerialName("ownerId")
    val ownerId: Long? = null,
    @SerialName("images")
    val images: List<String> = emptyList(),
    @SerialName("logoUrl")
    val logoUrl: String? = null,
    @SerialName("videoUrl")
    val videoUrl: String? = null,
    @SerialName("hasVideo")
    val hasVideo: Boolean = false,
    @SerialName("createdAt")
    val createdAt: String? = null
) {
    val firstImageUrl: String?
        get() = images.firstOrNull()?.let { getFullUrl(it) }

    val fullImageUrls: List<String>
        get() = images.map { getFullUrl(it) }

    fun getFullUrl(path: String): String = ApiClient.resolveUrl(path)
}
